commands/upgradeagents.go

```diff
@@ -185,6 +205,62 @@ func (c *upgradeAgentsImplCommand) getTools(ctx *cmd.Context, client *http.Clien
 	return nil
 }
 
+func (c *upgradeAgentsImplCommand) pushTools(ctx *cmd.Context, ver version.Number, scriptPath string, machines []FlatMachine) error {
+	var group errgroup.Group
+	for i := range machines {
+		machine := machines[i]
+		group.Go(func() error {
+			return errors.Annotatef(
+				c.pushToolsToMachine(ctx, ver, scriptPath, machine),
+				"machine %s", machine.ID)
+		})
+	}
+	logger.Debugf("waiting for copies to finish")
+	return group.Wait()
+}
+
+func (c *upgradeAgentsImplCommand) pushToolsToMachine(ctx *cmd.Context, ver version.Number, scriptPath string, machine FlatMachine) error {
+	logger.Debugf("making target dir for machine %s", machine.ID)
+	rc, err := runViaSSH(
+		machine.Address,
+		"rm -rf 1.25-agent-upgrade; mkdir 1.25-agent-upgrade; chown ubuntu:ubuntu 1.25-agent-upgrade",
+		withSystemIdentity())
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if rc != 0 {
+		return &cmd.RcPassthroughError{Code: rc}
+	}
+	toolsPath := path.Join(toolsDir, fmt.Sprintf("%s-%s", ver, seriesArch(machine)))
+	var options ssh.Options
+	options.SetIdentities(systemIdentity)
```

Comment:
```go
options.SetStrictHostKeyChecking(ssh.StrictHostChecksNo)
options.SetKnownHostsFile(os.DevNull)
```
?

or maybe just split out a function to get the base ssh/scp options, and use that both here and in runViaSSH?